service/telemetry/logger.go

```diff
@@ -37,23 +37,34 @@ func newLogger(set Settings, cfg Config) (*zap.Logger, log.LoggerProvider, error
 		return nil, nil, err
 	}
 
+	fields := []zap.Field{}
+	for k, v := range cfg.Resource {
+		if v != nil {
+			f := zap.Field{Key: k, Type: zapcore.StringType, String: *v}
+			fields = append(fields, f)
+		}
+	}
+
```

Comment:
```suggestion
	logger = logger.WithOptions(zap.WrapCore(func(c zapcore.Core) zapcore.Core {
		return c.With(fields)
	}))
```

See previous comment for rationale